### ISSUE TITLE
"Error in updating source" and "Problem getting file" errors

### ISSUE THREAD

**Issue by thenerdpaul** [StevenBlack/hosts#364]
In the latest version I get a bunch of these errors while running either the updateHostsFile.py script (the first one I tried) or makeHosts.py (which I tried later to see if it would fix the errors):

`Updating source extensions/fakenews from https://raw.githubusercontent.com/marktron/fakenews/master/fakenews
Problem getting file:  https://raw.githubusercontent.com/marktron/fakenews/master/fakenews
Error in updating source:  https://raw.githubusercontent.com/marktron/fakenews/master/fakenews`

**Comment by funilrys** [comment]
Strange issue ... It's working for me ...

**Comment by funilrys** [comment]
any feedback after 10 days @thenerdpaul ?

**Comment by thenerdpaul** [comment]
@funilrys: I tried a bunch of things (mostly I was hoping that it was an issue with my network), but no luck. Since I'm still getting an updated host file (even if it is lacking some things), I kind of just gave up. Hopefully (sort of?) someone more knowledgeable than me runs into a similar issue and knows how to fix it.

**Comment by thenerdpaul** [comment]
@StevenBlack: since this is still an issue...?

**Comment by funilrys** [comment]
Are you behind a proxy or a firewall? Those are the only thing I have in mind now ...

**Comment by thenerdpaul** [comment]
No to both.

**Comment by StevenBlack** [comment]
@thenerdpaul sorry, I presumed it appears clear this is a situation-local issue, specific to your circumstances.

Besides, you provide no details of OS, Python version, or anything else.

**Comment by thenerdpaul** [comment]
Okay, I can fix that.

Python version: Python 3.6.2 (v3.6.2:5fd33b5926, Jul 16 2017, 20:11:06).

OS: macOS 10.13 (17A315i) - Yes, this is a beta, but I highly doubt this is the issue.

Hardware: I am using a 12" MacBook (Early 2016; MacBook9,1).

My internet connection does not have a firewall that would block this script from fetching these files. Also, not that it really matters, but I have a pretty fast internet connection (compared to what I used to have at least - ~100Mbps down, ~50Mbps up) provided by Comcast with IPv4 & IPv6 fully enabled, so it shouldn't be an issue with it timing out or not being able to find the file. (I can load them manually after all). Also, it is worth noting that the issue still occurs even if I use the local coffee shop's WiFi network - though, I will admit, the coffee shop's WiFi probably has some sort of security system in place. (At least I would hope it does, for their sake).

I have this script set to run automatically every hour via cron. The above error occurs for all the additional categories whether I run the script manually the usual way or when it is run automatically. It's important to note, I think, that this problem does not occur for the main category - just the additional ones.

The script and all the related files have been moved to /etc/, but I either give it root privileges when I run the file manually, or it have been pre-given root privileges when it is run by cron. There's no issue when it comes to writing the files, in other words, considering it can fetch (and write to /etc/) the main file - again, only the additional ones have problems.

Anything else you would like to know?

**Comment by StevenBlack** [comment]
Thanks @thenerdpaul that's very helpful.

**Comment by prttygltch** [comment]
This was issue for me just now on macOS. I added code in the get_file_by_url() method to show the full exception, which was

`<urlopen error [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1056)>`

This was caused by Python not using the OpenSSL certificates from macOS. I ended up

1. Installing Python from python.org, instead of using the version that came installed on the Mac. If you're already using an installed (non-default) Python, skip this.
2. At the end of the install, it should prompt to run the `/Applications/Python 3.8/Install Certificates.command`. Do that.

Hopefully that solves the issue for you.
